pkg/health/health.go

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (C) 2022 IBM, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package health
+
+import (
+	"github.com/heptiolabs/healthcheck"
+	"github.com/netobserv/flowlogs2metrics/pkg/config"
+	log "github.com/sirupsen/logrus"
+	"net"
+	"net/http"
+	"time"
+)
+
+const defaultServerHost = "0.0.0.0"
+
+type Server struct {
+	handler healthcheck.Handler
+	address string
+}
+
+func (hs *Server) Serve() {
+	for {
+		err := http.ListenAndServe(hs.address, hs.handler)
+		log.Errorf("http.ListenAndServe error %v", err)
+		time.Sleep(60 * time.Second)
+	}
+}
+
+func NewHealthServer() *Server {
+
+	handler := healthcheck.NewHandler()
+	address := net.JoinHostPort(defaultServerHost, config.Opt.Health.Port)
+
+	handler.AddLivenessCheck("goroutine-threshold", healthcheck.GoroutineCountCheck(100))
+	handler.AddLivenessCheck("GCMaxPauseCheck", healthcheck.GCMaxPauseCheck(60*time.Second))
+	handler.AddReadinessCheck("HTTPGetCheck", healthcheck.HTTPGetCheck("https://quay.io", 60*time.Second))
```

Comment:
About the HTTP Get check, what if the pod is started in a disconnected environment?
Overall I think these health checks are interesting but I wonder if they could create more harm in some situations, even considering the two liveness checks above. We must keep in mind that when liveness probe is defined in kube, unhealthy pods are restarted, which is not always the best remedy to the symptoms.
So there are several options, we could make these checks more configurable, or simply being able to turn them on/off (which we can do just on the operator side, as it's where the kube probes are defined), or last option just remove them and keep health check more minimalist (ie. just return 200 when server is ready, no more)
Thoughts?

Reply:
I think in any case we should be able to turn off kube probes from the operator, through CR config

Reply:
@jotak I changed the code just to check that the "pipeline" is ready and alive based on a flag ... going forward we will add more and more validations that things are really alive and ready